Merge changes from mschnorb/VIC

mschnorb opened the pull request:
This request includes several feature updates, including 1) the addition of a second precipitation scaling factor (now scale precipitation separately for rain and snow), 2) addition of two output variables to capture changes in band elevation and band area (applicable during dynamic glacier simulations), and 3) an added variable to track glacier melt energy. Also included are corrections made to 1) the KIENZLE phase partitioning algorithm, 2) glacier energy balance calculation, and 3) the calculation of energy balance error.

mschnorb committed:
Corrected parameterization of KIENZLE routine for calculating rain as fraction of precipitation.

mschnorb committed:
Added two precipitation scaling parameters to scale rainfall (PADJ_R) and snowfall (PADJ_S) separately.

mschnorb committed:
Added variable to track glacier melt energy (i.e. erngy used to thaw glacier ice) and updated calculation of mass balance error for glacier HRUs; removed some inconsistencies between calculation of Glacier and Snowpack energy balance.

mschnorb committed:
Corrected conversion of IceDepth while calculating glacier energy balance so that it is non-recursive during iteration; corrected computation of snow surface energy balance components in solve_snow_glac.c (no uses correct value for snow coverage); removed ice density from calculation of glacier cold content and delta cold content.

mschnorb committed:
Minor format editing; commenting out debug code

mschnorb committed:
Merge branch 'padj_update'

mschnorb committed:
Added two output variables to capture changes in band elevation and band area during dynamic glacier simulations; fixed error in calculation of band glacier area

jameshiebert commented on GlacierEnergyBalance.c:
```diff
@@ -26,12 +25,13 @@ double GlacierEnergyBalance::calculate(double TSurf) {
   double Tmp;
   double VaporMassFlux;           /* Mass flux of water vapor to or from the intercepted snow (kg/m2s) */
   double Fbal;                    /* Energy balance at glacier surface */
+  double temp_IceDepth;			  /* Local variable to hold scaled value for IceDepth */
```
You might consider making this `const`

jameshiebert commented on SnowPackEnergyBalance.c:
```diff
@@ -90,16 +90,12 @@ double SnowPackEnergyBalance::calculate(double TSurf)
-				     (W/m2) */
-  double RestTerm;		  /* Rest term in surface energy balance
-				     (W/m2) */
+  /* double LongRadOut; */		  /* long wave radiation emitted by surface (W/m2) */
```
Please remove commented out non-code.

jameshiebert commented on StateIO.h:
```diff
@@ -468,8 +468,12 @@ enum StateMetaDataVariableIndices {
+  GLAC_MASS_BALANCE_INFO, // to store the Glacier Mass Balance information (grid cell num and equation) for use in VIC-RGM integration
+  NUM_GLAC_MASS_BALANCE_INFO_TERMS,
+
+  SOIL_PADJ_R, //new
```
Please don't put "//new" in the code. We have version control for a reason and can tell when things were added :)

jameshiebert commented on calc_water_energy_balance_errors.c:
```diff
@@ -54,6 +54,7 @@ void calc_energy_balance_error(int    rec,
 			       double sensible,
 			       double grnd_flux,
 			       double snow_fluxes,
+				   double glac_fluxes,
```
I think you have some whitespace mismatching here.

jameshiebert commented on glacier_melt.c:
```diff
@@ -84,6 +84,7 @@ int glacier_melt(double Le,
     double *save_Qnet,
     double *save_advection,
     double *save_deltaCC_glac,
+	double *save_glacier_melt_energy,
```
ditto

jameshiebert commented on initialize_atmos.c:
```diff
@@ -327,10 +327,9 @@ void initialize_atmos(atmos_data_struct        *atmos,
   /* Scale precipitation using PADJ */
-
-  for (int idx = 0; idx < (state->global_param.nrecs*state->NF); idx++) {
+  /* for (int idx = 0; idx < (state->global_param.nrecs*state->NF); idx++) {
```
Please remove commented out non-code.

mdfischer commented on surface_fluxes_glac.c:
```diff
@@ -238,10 +239,11 @@ int surface_fluxes_glac(
     Tair = atmos->air_temp[hidx] + soil_con->Tfactor[hru.bandIndex];
     step_prec[WET] = atmos->prec[hidx] / hru.mu * soil_con->Pfactor[hru.bandIndex];  // precipitation in mm
 
+    /** Calculate fraction of precipitation that falls as fain; scale rainfall and snowfall**/
```
You probably meant to write "rain" instead of "fain"

jameshiebert replied:
Still need to fix this typo.

mschnorb committed:
Mainly formatting changes, such as removing commented-out code and correcting whitespace mismatches.

mschnorb commented:
This commit addresses outstanding recommended changes and comments. Specifically, temp_IceDepth converted to constant, non-code removed (including some additional occurrences), removed "//new" comments (plus additional occurrences), corrected white space mismatches (including additional occurrences) and corrected spelling errors.

jameshiebert commented on surface_fluxes_glac.c:
```diff
@@ -281,6 +283,10 @@ int surface_fluxes_glac(
+    /* if (rec == 1411)
+  	  fprintf(stderr, "Record %i\n", rec); */
+
     if (step_snow.swq > 0. || snowfall[WET] > 0.) {
```
This non-code comment still needs to be removed.

jameshiebert commented:
Round 2 looks much better. Once those last two fixes are made, I'm happy for you guys to merge this.

mschnorb committed:
Removed more non-code.
